src/runtime-rs/crates/runtimes/common/src/container_manager.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::Result;
+use async_trait::async_trait;
+
+use crate::types::{
+    ContainerConfig, ExecProcessRequest, KillRequest, ProcessExitStatus, ProcessID,
+    ProcessStateInfo, ResizePtyRequest, ShimPid, ShutdownRequest, StatsInfo, UpdateRequest,
+};
+
+#[async_trait]
+pub trait ContainerManager: Send + Sync {
+    // container lifecycle
+    async fn create_container(&self, config: ContainerConfig) -> Result<u32>;
+    async fn pause_container(&self, cid: &str) -> Result<()>;
```

Comment:
At some later point, it might be nice to have a type for container ids, akin to `ProcessID`.

Reply:
Thanks for your review, add a type `ContainerID` for container id.